### PR TITLE
feat(database): encrypt repo fields for enhanced security

### PR DESCRIPTION
**Pull request by jbrockopp**
Dependent on https://github.com/go-vela/types/pull/163

This change will introduce encryption for repo fields that are stored in the Vela database.

We've decided to use the Advanced Encryption Standard (`AES`) with a `256` bit key (a.k.a. `AES-256`):

https://en.wikipedia.org/wiki/Advanced_Encryption_Standard

> NOTE: This change is backward compatible. If you try fetching an unencrypted repo, everything will work as expected.
>
> We'll plan to offer some sort of utility that will assist in encrypting all existing repos in the database.

### CODE REVIEW

File: database/repo.go [hunk 1]
```diff
@@ -161,6 +223,14 @@ func (c *client) CreateRepo(r *library.Repo) error {
 		return err
 	}
 
+	// decrypt the fields for the repo
```
Comment (davidvader) [inline database/repo.go:226]:
```suggestion
	// encrypt the fields for the repo
```

File: database/repo.go [hunk 2]
```diff
@@ -180,6 +250,14 @@ func (c *client) UpdateRepo(r *library.Repo) error {
...
```
Comment (davidvader) [inline database/repo.go:253]:
```suggestion
	// encrypt the fields for the repo
```

**Review by davidvader — CHANGES_REQUESTED** [review]
small typos

File: database/repo.go (hunk 1 above)
Comment (jbrockopp) [reply database/repo.go]: fixed 👍

File: database/repo.go (hunk 2 above)
Comment (jbrockopp) [reply database/repo.go]: fixed 👍

**Review by matt-fevold — APPROVED** [review]

**Review by davidvader — APPROVED** [review]

**Pull request merged**